src/utilities/sortDataBy.js

```diff
@@ -41,15 +41,25 @@ export const sortDataBy = (data, sortBy, sortDataType, isAscending = true) => {
 /**
  * Sorts an array of objects, returning a new array.
  * Sorts strings, numbers, dates or booleans.
- * Types: 'string', 'number', 'date', 'boolean'.
  * @param  {Array}   data          Array of objects to sort
  * @param  {String}  sortBy        Key for the field to sort by
- * @param  {String}  sortDataType  The type of data to sort
  * @param  {Boolean} isAscending   True if ascending, false otherwise.
  * @return {Array}   A new array of sorted data
  */
-export const newSortDataBy = (data, sortBy, sortDataType, isAscending = true) => {
-  switch (sortDataType) {
+export const newSortDataBy = (data, sortBy, isAscending = true, sortDataType) => {
+  const sortType =
+    sortDataType ||
+    {
+      otherPartyName: 'string',
+      itemCode: 'string',
+      itemName: 'string',
+      status: 'string',
+      serialNumber: 'number',
+      availableQuantity: 'number',
+      totalQuantity: 'number',
+    }[sortBy];
```

Comment:
this is going to end up massive (one for every column we use) - I think just defining a `const sortDataTypes` or something outside of the method might be better?

Reply:
Sure, although that doesn't fix it being huge. I was wondering if I should put it in the column definitions?

Reply:
That means the columns lookup table has to use the `column key` as the lookup table `key`, which means you can only ever have one column definition for a certain key. I don't think we use multiple column definitions at the moment, but it's pretty restrictive. I don't think it'd be tooooo big as an object in the module scope?

Reply:
Fair, I agree. https://github.com/openmsupply/mobile/pull/1164/commits/24f2a1ed1038dce3312805a32ec7966bc0e778c5